src/network/networkd-link.c

```diff
@@ -854,28 +854,26 @@ static int address_handler(sd_netlink *rtnl, sd_netlink_message *m, void *userda
 static int link_push_dns_to_dhcp_server(Link *link, sd_dhcp_server *s) {
         _cleanup_free_ struct in_addr *addresses = NULL;
         size_t n_addresses = 0, n_allocated = 0;
-        char **a;
+        unsigned i;
 
         log_debug("Copying DNS server information from %s", link->ifname);
 
         if (!link->network)
                 return 0;
 
-        STRV_FOREACH(a, link->network->dns) {
-                struct in_addr ia;
+        for (i = 0; i < link->network->n_dns; i++) {
```

Comment:
Commenting here, because github does not allow comments on the patch anymore...

> DNS servers must be specified as IP addresses

Is this true in general (not talking about current implementation atm)? Occasionally it is useful to specify a certain name in `/etc/hosts`, and then use that name to specify the dns resolvers. If `hosts` is before `dns`/`resolve` in `nsswitch.conf`, this works fine. Is this something we want to forbid?